src/odemis/acq/stream/_projection.py

```diff
@@ -1083,11 +1091,21 @@ class NeedRecomputeException(Exception):
         need_recompute = True
         while need_recompute:
             z = self._zFromMpp()
-            rect = self._rectWorldToPixel(self.rect.value)
+
+            # .rect was static, cant be updated when the image moves, we now use bounding box instead
+            # TODO: this works, but loads all tiles instead of ones in the view, might be good place to leave it
+            # TODO: get the bbox inside the .rect
+            view_rect = self._rectWorldToPixel(self.getBoundingBox())
```

Comment:
This changes quite a lot the behaviour of this function. Instead of returning just the tiles needed, it returns all the tiles. At z==0, that means loading the entire data. So the point of pyramidal/tiled image is lost.

`.rect` is not static. It can be updated live. Both its value *and* its .range can be updated. See `.clip_on_range = True` on the VA to simplify the update. This way, you'd adjust it whenever `force_image_update()` is called.
I guess that when this happens, you could reset its value to the whole range (as it's initialized). Then, it's up to the caller to recompute the new value. (alternatively, you could try to use the value and range prior to the update, and adjust the value linearly to the new range... but that might make things more complicated for not help).

Possibly, the other part that this change "solved" is when the image has (large) rotation. So far, `_rectWorldToPixel()` assumes there is no rotation. To handle rotation, you'd need to adjust this function to properly compute the bounding box. I think this can be done easily by just using the `Stream.getPixelCoordinates()` function on the four corner coordinates. (that function didn't exist when we wrote `_rectWorldToPixel()`, hence it's not currently using it).